codex/sales/slotqueue.nim

```diff
@@ -413,29 +335,26 @@ proc run(self: SlotQueue) {.async: (raises: []).} =
       if item.seen:
         trace "processing already seen item, pausing queue",
           reqId = item.requestId, slotIdx = item.slotIndex
-        self.pause()
+        queue.pause()
         # put item back in queue so that if other items are pushed while paused,
         # it will be sorted accordingly. Otherwise, this item would be processed
         # immediately (with priority over other items) once unpaused
         trace "readding seen item back into the queue"
-        discard self.push(item) # on error, drop the item and continue
-        worker.doneProcessing.complete()
-        if err =? self.addWorker().errorOption:
-          error "error adding new worker", error = err.msg
+        discard queue.push(item) # on error, drop the item and continue
         await sleepAsync(1.millis) # poll
         continue
 
       trace "processing item"
-
-      let fut = self.dispatch(worker, item)
-      self.trackedFutures.track(fut)
-
-      await sleepAsync(1.millis) # poll
+      without onProcessSlot =? queue.onProcessSlot:
+        error "slot queue onProcessSlot not set?"
+        continue
```

Comment:
Should we raise a `Defect` here instead? Seems like a pretty unrecoverable error to me. This how we handle callbacks not set elsewhere in the sales state machine.